### PR TITLE
Feature: bg gray link fix and a few other background color fixes

### PR DESCRIPTION
**Pull request by bspeare**
Resolves https://github.com/uiowa/uids/issues/651
Resolves https://github.com/uiowa/uids/issues/280
Resolves https://github.com/uiowa/uids/issues/592
Resolves https://github.com/uiowa/uids/issues/411

# How to test

## Test for blue link color within gray background sections

Go to https://uids.brand.uiowa.edu/branches/feature_bg_gray_link/components/preview/kitchen-sink.html and verify that:
- The link color is blue on gray background if link is within `p, ol, ul, dl, cite` and not using the `bttn` class - **can you think of any other tags we might want to target?**
- Look at the link color within the cards in all sections and the text outside of the card grid.  See screenshots below for examples.   
<details>
Screen Shot 2022-03-08 at 8 26 24 AM
Screen Shot 2022-03-08 at 8 26 38 AM
</details>

## Test for https://github.com/uiowa/uids/issues/651
- Blockquote borders remain gold on all background colors except gold if you look at each card on https://uids.brand.uiowa.edu/branches/feature_bg_gray_link/components/preview/kitchen-sink.html. 

## Test for https://github.com/uiowa/uids/issues/592
Go to https://uids.brand.uiowa.edu/branches/feature_bg_gray_link/components/preview/kitchen-sink.html 
- Navigate to a gold section and verify that `bttn--primary` is white text on black background. 
- Look at the `bttn--secondary` in a gold background section and verify that there is a button border on hover that makes the animated underline visible. 

## Test for https://github.com/uiowa/uids/issues/411

Go to https://uids.brand.uiowa.edu/branches/feature_bg_gray_link/components/preview/cta--gold.html
- Test with `bg--gold` or `bg--white` or `bg--gray` or `bg--black` on body class and verify contrast colors remain the same.

### CODE REVIEW

File: src/components/background/background.scss [hunk 1]
```diff
@@ -57,3 +57,11 @@
 [class*="bg--black"] {
   border-color: rgba(255,255,255,.325);
 }
+
+// override gray link color.
+[class*="bg--white"] :is(p, ul, ol, dl, cite) a,
+[class*="bg-"] [class*="bg--white"] :is(p, ul, ol, dl, cite) a,
+[class*="bg--gray"] :is(p, ul, ol, dl, cite) a,
+[class*="bg-"] [class*="bg--gray"] :is(p, ul, ol, dl, cite) a {
```
Comment (pyrello) [inline src/components/background/background.scss:65]: I'm wondering about the quotes here. I notice below that there are no quotes. Don't have a strong feeling either way, but want to be consistent.

Comment (bspeare) [reply src/components/background/background.scss:65]: Good point, maybe we should use quotes?  MDN docs use them https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors.

Comment (pyrello) [reply src/components/background/background.scss:65]: Sounds good to me.

**Comment by GaryRidgway** [comment]
I am going to try and test this, notes below.

1. I have noticed you cannot click links in blockquotes on a card.

**Review by GaryRidgway — APPROVED** [review]
All looks good, I will move the card overflow stuff to a separate PR because it is out of scope.

**Pull request merged**